### ISSUE TITLE
Different item object types are returned based on which API call you use.

### ISSUE THREAD

**Issue by BarnumD** [square/square-php-sdk#14]
As I've explained in this [forum post](https://developer.squareup.com/forums/t/customers-api-returning-different-object-types/451), the Square API is returning different object types depending on the type of call you make.  Additionally, it's returning different object types within the same call.

The following `listCatalog` function will return items of stdClass type.
```
function getSquareItems($type = 'ITEM'){
  $client = getApiClient();
  $catalogApi = $client->getCatalogApi();
  $cursor = 'initial';
  $items = array();

  while ($cursor){
    if ($cursor == 'initial'){$cursor = null;}
    $apiResponse = $catalogApi->listCatalog($cursor, $type);

    if ($apiResponse->isSuccess()) {
      $listCatalogResponse = $apiResponse->getResult();
      $items = array_merge ($items, $listCatalogResponse->getObjects());

      $cursor = $listCatalogResponse->getCursor();
    } else {
      $errors = $apiResponse->getErrors();
      sendNotification('Error in getSquareItems', 'ERROR');
      sendNotification($errors, 'ERROR');
    }
  }
  return $items;
}
```

Example output
```
Array
(
    [0] =>
stdClass Object
        (
            [type] => ITEM
            [id] => 54UILILDH27K7LTLLE54DNYS
            [updated_at] => 2020-07-11T00:09:58.626Z
            [version] => 1594426198626
            [is_deleted] => 
            [present_at_all_locations] => 1
            [item_data] => stdClass Object
                (
                    [name] => Test Item 1
                    [description] => test desc 1
                    [visibility] => PRIVATE
                    [tax_ids] => Array
                        (
                            [0] => Y3JU6BZL2EQUA4VRRDPSQTC4
                        )

                    [variations] => Array
                        (
                            [0] => stdClass Object
                                (
                                    [type] => ITEM_VARIATION
                                    [id] => M2L62Y3VUFCM2W3NGXCUS4N2
                                    [updated_at] => 2020-07-11T00:09:58.626Z
                                    [version] => 1594426198626
                                    [is_deleted] => 
                                    [present_at_all_locations] => 1
                                    [item_variation_data] => stdClass Object
                                        (
                                            [item_id] => 54UILILDH27K7LTLLE54DNYS
                                            [name] => Regular
                                            [sku] => asdf
                                            [ordinal] => 1
                                            [pricing_type] => FIXED_PRICING
                                            [price_money] => stdClass Object
                                                (
                                                    [amount] => 500
                                                    [currency] => USD
                                                )

                                            [location_overrides] => Array
                                                (
                                                    [0] => stdClass Object
                                                        (
                                                            [location_id] => ZE0YNMZC64NAD
                                                            [track_inventory] => 1
                                                        )

                                                )

                                            [track_inventory] => 1
                                        )

                                )

                        )

                    [product_type] => REGULAR
                    [skip_modifier_screen] => 
                )

        )
)
```

Meanwhile, the following `retrieveCatalogObject` function will create output with a different object type.
```
function getItem($objectId){
  $client = getApiClient();
  $catalogApi = $client->getCatalogApi();

  $apiResponse = $catalogApi->retrieveCatalogObject($objectId, false);

  if ($apiResponse->isSuccess()) {
      $retrieveCatalogObjectResponse = $apiResponse->getResult();
      $catalogObject = $retrieveCatalogObjectResponse->getObject();
  } else {
      $errors = $apiResponse->getErrors();
  }

  return $catalogObject;
}
```

Example output
```
Square\Models\CatalogObject Object
(
    [type:Square\Models\CatalogObject:private] => ITEM
    [id:Square\Models\CatalogObject:private] => 54UILILDH27K7LTLLE54DNYS
    [updatedAt:Square\Models\CatalogObject:private] => 2020-07-11T00:09:58.626Z
    [version:Square\Models\CatalogObject:private] => 1594426198626
    [isDeleted:Square\Models\CatalogObject:private] => 
    [customAttributeValues:Square\Models\CatalogObject:private] => 
    [catalogV1Ids:Square\Models\CatalogObject:private] => 
    [presentAtAllLocations:Square\Models\CatalogObject:private] => 1
    [presentAtLocationIds:Square\Models\CatalogObject:private] => 
    [absentAtLocationIds:Square\Models\CatalogObject:private] => 
    [imageId:Square\Models\CatalogObject:private] => 
    [itemData:Square\Models\CatalogObject:private] => Square\Models\CatalogItem Object
        (
            [name:Square\Models\CatalogItem:private] => Test Item 1
            [description:Square\Models\CatalogItem:private] => test desc 1
            [abbreviation:Square\Models\CatalogItem:private] => 
            [labelColor:Square\Models\CatalogItem:private] => 
            [availableOnline:Square\Models\CatalogItem:private] => 
            [availableForPickup:Square\Models\CatalogItem:private] => 
            [availableElectronically:Square\Models\CatalogItem:private] => 
            [categoryId:Square\Models\CatalogItem:private] => 
            [taxIds:Square\Models\CatalogItem:private] => Array
                (
                    [0] => Y3JU6BZL2EQUA4VRRDPSQTC4
                )

            [modifierListInfo:Square\Models\CatalogItem:private] => 
            [variations:Square\Models\CatalogItem:private] => Array
                (
                    [0] => stdClass Object
                        (
                            [type] => ITEM_VARIATION
                            [id] => M2L62Y3VUFCM2W3NGXCUS4N2
                            [updated_at] => 2020-07-11T00:09:58.626Z
                            [version] => 1594426198626
                            [is_deleted] => 
                            [present_at_all_locations] => 1
                            [item_variation_data] => stdClass Object
                                (
                                    [item_id] => 54UILILDH27K7LTLLE54DNYS
                                    [name] => Regular
                                    [sku] => asdf
                                    [ordinal] => 1
                                    [pricing_type] => FIXED_PRICING
                                    [price_money] => stdClass Object
                                        (
                                            [amount] => 500
                                            [currency] => USD
                                        )

                                    [location_overrides] => Array
                                        (
                                            [0] => stdClass Object
                                                (
                                                    [location_id] => ZE0YNMZC64NAD
                                                    [track_inventory] => 1
                                                )

                                        )

                                    [track_inventory] => 1
                                )

                        )

                )

            [productType:Square\Models\CatalogItem:private] => REGULAR
            [skipModifierScreen:Square\Models\CatalogItem:private] => 
            [itemOptions:Square\Models\CatalogItem:private] => 
        )

    [categoryData:Square\Models\CatalogObject:private] => 
    [itemVariationData:Square\Models\CatalogObject:private] => 
    [taxData:Square\Models\CatalogObject:private] => 
    [discountData:Square\Models\CatalogObject:private] => 
    [modifierListData:Square\Models\CatalogObject:private] => 
    [modifierData:Square\Models\CatalogObject:private] => 
    [timePeriodData:Square\Models\CatalogObject:private] => 
    [productSetData:Square\Models\CatalogObject:private] => 
    [pricingRuleData:Square\Models\CatalogObject:private] => 
    [imageData:Square\Models\CatalogObject:private] => 
    [measurementUnitData:Square\Models\CatalogObject:private] => 
    [itemOptionData:Square\Models\CatalogObject:private] => 
    [itemOptionValueData:Square\Models\CatalogObject:private] => 
    [customAttributeDefinitionData:Square\Models\CatalogObject:private] => 
    [quickAmountsSettingsData:Square\Models\CatalogObject:private] => 
)
```
You can see above that this actually returned by Square CatalogItem fields as well as stdClass fields in the same object.

**But that's not the only issue.**  It seems that when the Square API returns a CatalogItem field, it returns fields with no underscores in their name and when it returns a stdClass object, it has underscores.  For instance, one of the results above has `itemData`, while the other has `item_data`.  Unfortunately my fix that I mentioned in the forum posting of casting it to the stdClass type won't work because you're providing me with completely different field names depending on which API endpoints I call.

The same issue of mixed object types exists in the return object of `upsertCatalogObject`.

**Comment by sseaman** [comment]
Thanks for reporting this.  We are looking into this now.

**Comment by BarnumD** [comment]
I believe this is all resolved in `6.1.0.20200722`
